samples/vboxwrapper/vbox_common.cpp

```diff
@@ -262,51 +283,206 @@ void VBOX_BASE::dump_hypervisor_logs(bool include_error_logs) {
+		    hardware acceleration for virtual machines, but your processor does not support the required feature.\n \
+		    Please report this issue to the project so that it can be addresssed.\n",
+		
+		"   NOTE: VirtualBox hypervisor reports that another hypervisor has locked the hardware acceleration\n \
+		    for virtual machines feature in exclusive mode.\n",
+		
+		"   NOTE: VirtualBox has failed to allocate enough memory to start the configured virtual machine.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+		
+		"   NOTE: VM failed to enter an online state within the timeout period.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+	
+		    "VM environment needed to be cleaned up.",
+       
+		    "Forign VM Hypervisor locked hardware acceleration features.",
+
+		    "VM Hypervisor was unable to allocate enough memory to start VM.",
+
+	     	"VM Hypervisor failed to enter an online state in a timely fashion."
+	};
+
+	std::vector<std::string> v(args, args + 14);
+	
+	return v[num];
+}
+
+string VBOX_BASE::read_vm_log(){
+	string line;
+	size_t line_pos;
+	size_t line_end;
+	size_t line_start;
+	string msg;
+	string virtualbox_vm_log;
+	virtualbox_vm_log = vm_master_name + "/Logs/VBox.log";
+	
+
+	std::ifstream  src(virtualbox_vm_log.c_str(), std::ios::binary);
+	    
+	if ((src.is_open()) && (src.seekg(log_pointer))) {
+
+            while (std::getline(src, line)) {
+
+                line_start = line.find("Console:");
+		if (line_start != string::npos) {
+					
+                    line_start += 35;
+                    line_end = line.find("\'", line_start);
+                    msg = line.substr(line_start, line_end - line_start);
+                    sanitize_format(msg);
+                    state = msg.c_str();
+                }
+                line_pos = line.find("Guest Log:");
+                if (line_pos != string::npos) {
+					
+                    msg = line.substr(line_pos, line.size() - line_pos);
+                    sanitize_format(msg);
+                    vboxlog_msg(msg.c_str());
+                }
+                log_pointer = src.tellg();
+           }
+           return state.c_str();
+        }
+        else return "Error in parsing the log file";
+}
+
+
+// Dump any new guest log messages which are generated by applications running within
+// the guest VM.
+void VBOX_BASE::dump_vmguestlog_entries() {
+    string local_vm_log;
+    string line;
+    size_t line_pos;
+    VBOX_TIMESTAMP current_timestamp;
+    string msg;
+	string virtualbox_vm_log;
+	virtualbox_vm_log = vm_master_name + "/Logs/VBox.log";
+
+	if (boinc_file_exists(virtualbox_vm_log.c_str())) {
+
+		std::ifstream  src(virtualbox_vm_log.c_str(), std::ios::binary);
+		while (std::getline(src, line))
+		{
+			line_pos = line.find("Guest Log:");
+			if (line_pos != string::npos) {
+				sscanf(
```

Comment:
I would check the return value of `sscanf`. Otherwise you risk setting `vm_log_timestamp` to garbage.